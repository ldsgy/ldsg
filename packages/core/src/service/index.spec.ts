import { Service } from ".";

describe("service", () => {
  test("getUriInfo", async () => {
    const service = new Service({
      settings: {
        uri: "uri://user:pass@example.com:123/one/two.three?q1=a1&q2=a2#body",
      },
    });

    expect(service.getUriInfo()).toMatchSnapshot();
  });
});
