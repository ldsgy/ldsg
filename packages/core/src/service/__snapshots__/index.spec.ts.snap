// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`service getUriInfo 1`] = `
{
  "fragment": "body",
  "host": "example.com",
  "path": "/one/two.three",
  "port": 123,
  "query": "q1=a1&q2=a2",
  "reference": "uri",
  "scheme": "uri",
  "userinfo": "user:pass",
}
`;
